### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by amar-josh**
Search Feature API integration.
https://peerly-josh.atlassian.net/browse/PEER-28

### CODE REVIEW

File: src/app/screens/PeerlyScreens/ProfileDetailScreen/profile.hooks.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@ import toast from '../../../utils/toast';
 
 import {getProfileDetails} from '../../../services/PeerlyServices/profileDetail';
 
-export function useGetProfileDetails() {
+export function useGetProfileDetails(userId?: number) {
   const {data, isLoading, isFetching, isSuccess, isError} = useQuery({
-    queryKey: ['profile_icon'],
+    queryKey: userId ? ['profile_icon', userId] : ['profile_icon'],
```
Comment (Chetan-Satpute-Josh) [inline src/app/screens/PeerlyScreens/ProfileDetailScreen/profile.hooks.ts:10]: `queryKey: ['profile_icon', userId]`

Should also work in this case.

**Review by Chetan-Satpute-Josh — CHANGES_REQUESTED** [review]

File: src/app/screens/PeerlyScreens/DetailsScreen/index.tsx [hunk 2]
```diff
@@ -90,7 +104,7 @@ const AppreciationDetailsScreen = ({route}) => {
       <View style={{justifyContent: 'center', alignItems: 'center'}}>
         <View style={styles.info}>
           <Text style={styles.name}>
-            {`${cardDetails.receiver_first_name} ${cardDetails.receiver_last_name} `}
+            {`${cardDetails?.receiver_first_name} ${cardDetails?.receiver_last_name} `}
```
[the remark this answers is not in the document]
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/DetailsScreen/index.tsx:107]: Resolve with add empty string ""

File: src/app/screens/PeerlyScreens/HomeScreen/index.tsx [hunk 3]
```diff
@@ -112,24 +126,29 @@ const HomeScreen = ({navigation}) => {
...
       </Pressable>
-
-      <TextInput style={styles.searchInput} placeholder={'Search Co-Worker'} />
-
+      <View>
+        <TextInput
+          onPressIn={handleSearchPress}
+          style={styles.searchInput}
+          value={''}
+          placeholder={'Search Co-Worker'}
```
[the remark this answers is not in the document]
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/HomeScreen/index.tsx:149]: Updated

File: src/app/screens/PeerlyScreens/HomeScreen/index.tsx [hunk 4]
```diff
@@ -52,26 +56,32 @@ const HomeScreen = ({navigation}) => {
...
-      <FlatList
-        data={activeUsersList}
-        renderItem={({item}) => <LeaderBoardCard userDetail={item} />}
-        keyExtractor={item => item.id}
-        horizontal={true}
-      />
-    </View>
+  const FirstRoute = useCallback(
+    () => (
+      <View style={{flex: 1, backgroundColor: '#F4F6FF'}}>
```
[the remark this answers is not in the document]
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/HomeScreen/index.tsx:61]: In UI Beautification which will be the next PR we will remove all inline styles of code.

File: src/app/screens/PeerlyScreens/SearchScreen/types.ts [hunk 5]
```diff
@@ -1 +1 @@
-export interface demo {}
+
```
[the remark this answers is not in the document]
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/SearchScreen/types.ts:1]: Removed the file

File: src/app/screens/PeerlyScreens/components/GivenAndReceivedAppreciation.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    ),
+    [expressedList, handleAppreciationCardClick],
+  );
+
+  const renderScene = SceneMap({
+    received: FirstRoute,
+    expressed: SecondRoute,
+  });
+
+  const renderTabBar = (props: any) => (
```
[the remark this answers is not in the document]
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/components/GivenAndReceivedAppreciation.tsx:104]: Updated with native tab bar types.

File: src/app/screens/PeerlyScreens/navigation/index.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import {RootStackParamList} from './types';
+
+const RootStack = createNativeStackNavigator<RootStackParamList>();
+
+function PeerlyHomeScreen() {
+  const {isLoading, isFetching, isError} = useLoginPeerly();
+  if (isLoading || isFetching) {
+    return (
+      <View>
+        <ActivityIndicator size={'large'} />
```
[the remark this answers is not in the document]
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/navigation/index.tsx:27]: Done

File: src/app/screens/PeerlyScreens/ProfileDetailScreen/profile.hooks.ts (hunk 1 above)
Comment (amar-josh) [reply src/app/screens/PeerlyScreens/ProfileDetailScreen/profile.hooks.ts]: Made the chagnes

**Review by Chetan-Satpute-Josh — APPROVED** [review]

**Pull request merged**